canvas_handler.py

```diff
@@ -115,13 +115,7 @@ def _ids_converter(ids: Tuple[str, ...]) -> List[int]:
             List of int ids
         """
 
-        temp = []
-
-        for i in ids:
-            temp.append(int(i))
-
-        temp = list(dict.fromkeys(temp))
```

Comment:
I don't even know what was going on here. Creating a list, then a dict from the list, then a list again. This is equivalent to creating a list, then a set from the list, then creating a list again. Better to just start with a set. Then I checked usage and it actually made more sense to keep the set than convert to a list.